libretro-common/dynamic/dylib.c

```diff
@@ -34,6 +34,270 @@
 #include <dlfcn.h>
 #endif
 
+#if defined(__linux__) && !defined(ANDROID)
```

Comment:
This goes a bit too far for me. Hide this under a suitable ifdef. Don't enable this by default.